### PR TITLE
[4.1.0][CC] Exposing a SOAP endpoint as a REST API in Choreo Connect

### PR DESCRIPTION
**Pull request by VajiraPrabuddhaka**
## Purpose

This PR introduces the docs for how to expose a SOAP endpoint as a REST API in Choreo Connect.

### CODE REVIEW

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
+# Exposing a SOAP Endpoint as a REST API
```
Comment (VajiraPrabuddhaka) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md:1]: Change title as "REST API from a SOAP Endpoint"

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+For more information refer to [Creating an Integration Project]({{base_path}}/integrate/develop/create-integration-project/).
+
+### Step 2 - Create the REST API
+
+1. In the left hand side project explorer panel, expand the created project `PhoneVerification`.
+2. Right click on `PhoneVerfificationConfigs` and select **New->REST API**.
+3. Select the **Import API Artifact** option and provide file having content of [this](link from mgw sample project).
+
+!!! Note
```
Comment (VajiraPrabuddhaka) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md:46]: mention about the sample

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+    ```url tab="Linux"
+    http://172.17.0.1:8290/phoneverify
+    ```
+
+    ```url tab="MacOS/Windows"
+    http://host.docker.internal:8290/phoneverify
+    ```
+
+#### Update Query Parameters in Swagger definition
```
Comment (VajiraPrabuddhaka) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md:107]: can remove this

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+1. Log in to Developer Portal via `https://localhost:9444/devportal` and select the newly created API.
+2. Navigate to the **Subscriptions** page and subscribe the API to the default application visible as **DefaultApplication** with an available Rate Limiting Policy.
+3. Now go to the **Applications** and select **DefaultApplication**.
+4. Click on **Production Keys** and then click **Generate Keys** with the default settings, to generate production keys.
+5. Now go back to APIs and select your API and go to **Try Out** section from there. In the Try Out section you can get a Test Key with the **DefaultApplication** for production endpoint by clicking on **GET TEST KEY** .
+6. Then you can invoke the API using the Swagger console UI that appears in the same page or you may use `cURL` to invoke it. A sample `cURL` looks like below.
+    ```bash
+    curl -X 'GET' 'https://localhost:9095/phoneverify/1.0.0/checkphonenumber?PhoneNumber=8006785432&LicenseKey=8006785432' -H 'accept: */*' -H 'Authorization: Bearer <generated_key>' -k
+    ```
+
```
Comment (VajiraPrabuddhaka) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md:223]: add a note pointing to more information

Comment (VajiraPrabuddhaka) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md:200]: This is not required as the guide describes this itself.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md (hunk 3 above)
Comment (VajiraPrabuddhaka) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md]: fixed via https://github.com/wso2/docs-apim/pull/6114/commits/d03a84262c3c0fa449b465e58f50d312a1220157

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md (hunk 2 above)
Comment (VajiraPrabuddhaka) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md:56]: Fixed via https://github.com/wso2/docs-apim/pull/6114/commits/d03a84262c3c0fa449b465e58f50d312a1220157

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md (hunk 1 above)
Comment (VajiraPrabuddhaka) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md]: fixed via https://github.com/wso2/docs-apim/pull/6114/commits/d03a84262c3c0fa449b465e58f50d312a1220157

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
+# REST API from a SOAP Endpoint
+
+When it comes to web services, they are designed to provide rich functionlity to end users by supporting interoperable interactions over a network. Web services are mainly categorized into two types called SOAP and RESTful services. However due to various reasons like flexibility, scalability, complexity, performance, etc. RESTful services became more better for modern clients. Due to this reason exposing a SOAP endpoint as a RESTfull service is helpfull as it provides more flexibilities when integrating web services with various end user applications.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md:3]:
```suggestion
When it comes to web services, they are designed to provide rich functionlity to end users by supporting interoperable interactions over a network. Web services are mainly categorized into two types called SOAP and RESTful services. However, due to various reasons like flexibility, scalability, complexity, performance, etc. RESTful services became better for modern clients. Due to this reason exposing a SOAP endpoint as a RESTfull service is helpful as it provides more flexibilities when integrating web services with various end user applications.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+#### Step 1 - Initialize the apictl project
+
+Use the follwing command to initialize the `apictl` project.
+
+```
+apictl init PhoneVerify --oas http://localhost:8290/PhoneVerify?swagger.json
+```
+
+!!! Note
+    The swagger URL `http://localhost:8290/PhoneVerify?swagger.json` is used to get the swagger definition of the deployed integration service in previous steps. You may read more details on this from [Using Swagger Documents]({{base_path}}/integrate/develop/advanced-development/using-swagger-for-apis/).
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md:219]:
```suggestion
    The Swagger URL `http://localhost:8290/PhoneVerify?swagger.json` is used to get the Swagger definition of the deployed integration service in previous steps. You may read more details on this from [Using Swagger Documents]({{base_path}}/integrate/develop/advanced-development/using-swagger-for-apis/).
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+
+    <img src="{{base_path}}/assets/img/deploy/mgw/apim-service-catalog.png" alt="SOAP to REST request flow" width="650px" height="350px"/>
+
+2. Click on the PhoneVerify service and then click on **Create API** in the next window and create the API.
+
+    <img src="{{base_path}}/assets/img/deploy/mgw/apim-create-api-using-service-catalog.png" alt="SOAP to REST request flow" width="650px"/>
+
+#### Step 2 - Deploy the API using API Manager Publisher Portal
+
+Go to the **Deployments** from the left side panel of the Publisher portal and Deploy the API.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-rest-to-soap-api.md:183]:
```suggestion
Go to the **Deployments** from the left side panel of the Publisher Portal and Deploy the API.
```

**Review by pubudu538 — APPROVED** [review]

**Review by samgnaniah — APPROVED** [review]

**Pull request merged**